lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -407,22 +409,35 @@ putStakeCertificate
     -> ChimericAccount
     -> Put
 putStakeCertificate (PoolId poolId) (ChimericAccount accId) = do
+    assertLength "poolId" poolId 32
+    assertLength "account pubKey" accId 32
     putByteString accId
     putWord8 (stakeDelegationTypeTag DlgFull)
     putByteString poolId
+  where
+    assertLength desc x expected =
+        unless (BS.length x == expected) $
+            fail $ desc ++ " should be of length " ++ show expected
+                ++ " but has length " ++ show (BS.length x)
 
 putStakeDelegationTx
-    :: PoolId
-    -> ChimericAccount
-    -> Hash "AccountSignature"
+    :: Put -- ^ Payload
     -> [(TxIn, Coin)]
     -> [TxOut]
     -> [TxWitness]
+    -> (ByteString -> ByteString)
     -> Put
-putStakeDelegationTx poolId accId (Hash accSig) inputs outputs witnesses = do
-    putStakeCertificate poolId accId
-    putSignedTx mempty inputs outputs witnesses
-    putByteString accSig
+putStakeDelegationTx payload inputs outputs witnesses auth = do
+    withHeader MsgTypeDelegation $ do
+        withAppendedSignature auth $ do
+            putSignedTx payload inputs outputs witnesses
+
+withAppendedSignature :: (ByteString -> ByteString) -> Put -> Put
```

Comment:
why not to put it in where of `putStakeDelegationTx` if we do not use it anywhere else?